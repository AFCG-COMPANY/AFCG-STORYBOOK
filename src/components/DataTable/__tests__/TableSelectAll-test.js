import React from 'react';
import { mount } from 'enzyme';
import { Table, TableHead, TableRow, TableSelectAll } from '../';

describe('DataTable.TableSelectAll', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      id: 'id',
      name: 'select-all',
      checked: false,
      onSelect: jest.fn(),
      className: 'custom-class-name',
    };
  });

  it('should render', () => {
    const wrapper = mount(
      <Table>
        <TableHead>
          <TableRow>
            <TableSelectAll {...mockProps} />
          </TableRow>
        </TableHead>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke `onSelect` when clicked', () => {
    const wrapper = mount(
      <Table>
        <TableHead>
          <TableRow>
            <TableSelectAll {...mockProps} />
          </TableRow>
        </TableHead>
      </Table>
    );
    wrapper.find('InlineCheckbox input').simulate('click');
    expect(mockProps.onSelect).toHaveBeenCalledTimes(1);
  });
});
