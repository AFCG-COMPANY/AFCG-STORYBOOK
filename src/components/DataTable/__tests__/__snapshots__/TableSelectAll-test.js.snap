// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable.TableSelectAll should render 1`] = `
<Table>
  <table
    className="bx--data-table-v2 bx--data-table-v2--zebra"
  >
    <TableHead>
      <thead>
        <TableRow>
          <tr>
            <TableSelectAll
              ariaLabel="Select all rows in the table"
              checked={false}
              className="custom-class-name"
              id="id"
              name="select-all"
              onSelect={[MockFunction]}
            >
              <th
                scope="col"
              >
                <InlineCheckbox
                  ariaLabel="Select all rows in the table"
                  checked={false}
                  id="id"
                  name="select-all"
                  onClick={[MockFunction]}
                >
                  <input
                    checked={false}
                    className="bx--checkbox"
                    id="id"
                    name="select-all"
                    onClick={[MockFunction]}
                    type="checkbox"
                  />
                  <label
                    aria-label="Select all rows in the table"
                    className="bx--checkbox-label"
                    htmlFor="id"
                  />
                </InlineCheckbox>
              </th>
            </TableSelectAll>
          </tr>
        </TableRow>
      </thead>
    </TableHead>
  </table>
</Table>
`;
